opentelemetry-api/src/opentelemetry/trace/__init__.py

```diff
@@ -70,6 +70,47 @@
 ParentSpan = typing.Optional[typing.Union["Span", "SpanContext"]]
 
 
+class Link:
+    """A link to a `Span`."""
+
+    def __init__(
+        self, context: "SpanContext", attributes: types.Attributes = None
+    ) -> None:
+        self._context = context
+        self._attributes = attributes
+
+    @property
+    def context(self) -> "SpanContext":
+        return self._context
+
+    @property
+    def attributes(self) -> types.Attributes:
+        return self._attributes
+
+
+class Event:
+    """A text annotation with a set of attributes."""
+
+    def __init__(
+        self, name: str, timestamp: int, attributes: types.Attributes = None
+    ) -> None:
+        self._name = name
```

Comment:
A general question - do we expect this to be implemented in the API package or the SDK?
I think it makes sense to have it the API package, and I don't expect people to override this in the SDK.
Just to put the question here to see if someone has a different understanding.

Reply:
I'd +1 for keeping it in the API.

Reply:
By keeping it on the API we are providing a reference implementation and also allowing users to extend it if needed.

Reply:
> I think it makes sense to have it the API package

Agreed, that's how we do it in Java too, btw.